lib/smack/SmackInstGenerator.cpp

```diff
@@ -798,9 +798,9 @@ void SmackInstGenerator::visitCallInst(llvm::CallInst &ci) {
     emit(rep->call(f, ci));
   }
 
-  if (f->isDeclaration() && rep->isExternal(&ci)) {
+  if (f->isDeclaration()) {
     std::string name = naming->get(*f);
-    if (!EXTERNAL_PROC_IGNORE.match(name))
+    if (!EXTERNAL_PROC_IGNORE.match(name) && rep->isExternal(&ci))
```

Comment:
What does this line have to do with region collapse?

Reply:
Function `isExternal` under the hood creates a region, which can then lead to region collapses.

My first attempt at resolving this issue was to rewrite `isExternal` to not create a region, but that lead to other problems, in particular to this one: https://github.com/smackers/smack/issues/473#issuecomment-551199240